### PR TITLE
Fetch hosting alias correctly.

### PR DESCRIPTION
**Pull request by grasmash**

### CODE REVIEW

File: src/Command/CommandBase.php [hunk 1]
```diff
@@ -1089,7 +1089,7 @@ public static function getAliasCache() {
    */
   protected function doGetApplicationFromAlias($application_alias) {
     $acquia_cloud_client = $this->cloudApiClientService->getClient();
-    $acquia_cloud_client->addQuery('filter', 'hosting=@*' . $application_alias);
+    $acquia_cloud_client->addQuery('filter', 'hosting=prod:' . $application_alias);
```
Comment (grasmash) [inline src/Command/CommandBase.php:1092]: Sometimes this wildcard causes multiple applications to be returned. E.g., when an alias is `blue` and there are apps `prod:greenblue, prod:veryblue, prod:blue`, then `prod:greenblue` may be returned when `blue` is requested.

Comment (grasmash) [reply src/Command/CommandBase.php:1092]: Trouble is this may break ACSF sites. Not sure those actually start with `prod:`

Comment (grasmash) [reply src/Command/CommandBase.php:1092]: Yeah, possible prefix values are devcloud, gardens, prod, network, enterprise-g1.

File: src/Command/CommandBase.php [hunk 2]
```diff
@@ -1089,7 +1089,13 @@ public static function getAliasCache() {
    */
   protected function doGetApplicationFromAlias($application_alias) {
     $acquia_cloud_client = $this->cloudApiClientService->getClient();
-    $acquia_cloud_client->addQuery('filter', 'hosting=prod:' . $application_alias);
+    $possible_hosting_prefixes = ['prod', 'devcloud', 'network', 'enterprise-g1', 'gardens'];
```
Comment (anavarre) [inline src/Command/CommandBase.php:1092]: Looks like you've already figured this out but FYI you can find the info for any sub with:

```
acli api:applications:find <UUID> | jq -r '.hosting.id' | tr -d \" | sed 's/:.*//'
```

**Comment by danepowell** [comment]
If I understand correctly, the problem is that we were querying Cloud API applications using a wildcard filter, which meant that an alias of "foo" could return "openfoo"?

Instead of enumerating the hosting realms, wouldn't it be easier to change the filter to `hosting=@*:`? I tested briefly and this seems to work.

File: src/Command/CommandBase.php [hunk 3]
```diff
@@ -1089,7 +1089,8 @@ public static function getAliasCache() {
    */
   protected function doGetApplicationFromAlias($application_alias) {
     $acquia_cloud_client = $this->cloudApiClientService->getClient();
-    $acquia_cloud_client->addQuery('filter', 'hosting=@*' . $application_alias);
+    $filter_value = self::createHostingIdParamValue($application_alias);
+    $acquia_cloud_client->addQuery('filter', 'hosting=' . $filter_value);
```
Comment (danepowell) [inline src/Command/CommandBase.php:1093]:
```suggestion
    $acquia_cloud_client->addQuery('filter', 'hosting=@*:' . $filter_value);
```

Just changing this (and nothing else) seems to fix the issue for me.

**Comment by grasmash** [comment]
> If I understand correctly, the problem is that we were querying Cloud API applications using a wildcard filter, which meant that an alias of "foo" could return "openfoo"?
> 
> Instead of enumerating the hosting realms, wouldn't it be easier to change the filter to `hosting=@*:`? I tested briefly and this seems to work.

🤦

**Pull request closed without merging**